### DISCUSSION TITLE
change apt to nala

### DISCUSSION THREAD

**Discussion by root9191** [MichaIng/DietPi#6899] [Q&A]
Can i change the package manager to nala also for dietpi commands?

**Comment by Joulinar** [comment]
We use apt-get and there is no variable to change this. I guess answer would be: No

**Comment by MichaIng** [comment]
What is the reason you want to have `nala` support integrated?

For the UI enhancements, it of course does not make any sense, since the `G_AG*` commands are non-interactive anyway. For the parallel downloads part: There was `apt-fast` support in the past, but doing own tests, it was actually slower instead of faster, compared to `apt-get` with the global Debian CDN:
- https://github.com/MichaIng/DietPi/issues/1987#issuecomment-410004392
- https://github.com/MichaIng/DietPi/issues/2817
- https://github.com/ilikenwf/apt-fast/issues/163

**Reply by root9191** [reply]
In my opinion nala does a better and faster job than apt.
So also in future no possibility to change the package manager?

**Reply by MichaIng** [reply]
First of all nala is an `apt` frontend/wrapper, not a replacement. Since the UI enhancements are not relevant for the non-interactive calls in our scripts, the "faster" aspect is the only arguments. And that is not so much an opinion, bit either it indeed does APT installs/upgrades significantly faster or not. Did you make benchmarks to verify that, compared to plain `apt` with the global `deb.debian.org` CDN? As said, at first it sounds logic that parallel downloads are faster, but the `apt-fast` example showed that this is true only in rare edge  cases, while the additional processing in fact slows down the whole installation in the majority cases. A small benefit for a small number of users is IMO not worth the effort to implement and maintain support for additional APT tools.

Of course this does not mean that you cannot install and use nala for your own/manual APT installs, just out scripts do not use it.

And also if someone else does the implementation, we might accept it. Just my own time is limited and there are much more important tasks.
